File: 每日一题/637. 二叉树的层平均值.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def averageOfLevels(self, root: TreeNode) -> List[float]:
        ret = []
        q = [root]
        while(len(q)):
            size = len(q)
            temp_sum = 0
            for i in range(size):
                top = q.pop(0)
                temp_sum += top.val
                if top.left:
                    q.append(top.left)
                if top.right:
                    q.append(top.right)
            ret.append(temp_sum/size)
        return ret

File: 每日一题/637. 二叉树的层平均值.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution
{
public:
    vector<double> averageOfLevels(TreeNode *root)
    {
        auto q = queue<TreeNode *>();
        auto ret = vector<double>();
        q.push(root);
        while (!q.empty())
        {
            double sum = 0;
            auto size = q.size();
            for (int i = 0; i < size; i++)
            {
                auto top = q.front();
                q.pop();
                sum += top->val;
                if (top->left)
                {
                    q.push(top->left);
                }
                if (top->right)
                {
                    q.push(top->right);
                }
            }
            ret.push_back(sum / size);
        }
        return ret;
    }
};